hpoflow/optuna_transformers.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2021 Timothy Wolff-Piggott
+# This software is distributed under the terms of the MIT license
+# which is available at https://opensource.org/licenses/MIT
+
+import logging
+import os
+
+import mlflow
+
+from hpoflow.optuna_mlflow import OptunaMLflow
+from optuna._imports import try_import
+
+with try_import() as _imports:
+    import transformers
+_imports.check()
```

Comment:
I think this `_imports.check()` should not be called here but later when the class is actualy used (in the `__init__` function).

Reply:
This was my original implementation but it ends up being a bit more complicated, because the `OMLflowCallback` class needs to inherit from `transformers.TrainerCallback`, which may or may not be available.

https://github.com/PhilipMay/mltb/pull/53#issuecomment-825796849

I can switch back to that original, but we previously decided to favour the simpler implementation

Reply:
This is the lazy import implementation:

https://github.com/PhilipMay/mltb/pull/53/commits/73e9bbfd6d7de32a990607e2c7dda332097547da

Reply:
Ahh yes. I remember the discussion. Sorry. So this is ok.

Reply:
@twolffpiggott maybe add a comment that explains this? like this:

`# do the check here and not in the constructor because the class inherits from transformers.TrainerCallback`